vcsutils/webhookparser/github.go

```diff
@@ -1,41 +1,62 @@
 package webhookparser
 
 import (
+	"context"
 	"errors"
+	"fmt"
 	"net/http"
 	"strings"
 
 	"github.com/google/go-github/v45/github"
+
+	"github.com/jfrog/froggit-go/vcsclient"
 	"github.com/jfrog/froggit-go/vcsutils"
 )
 
 // GitHubWebhook represents an incoming webhook on GitHub
 type GitHubWebhook struct {
-	request *http.Request
+	logger vcsclient.Log
+	// used for GitHub On-prem
+	endpoint string
 }
 
 // NewGitHubWebhook create a new GitHubWebhook instance
-func NewGitHubWebhook(request *http.Request) *GitHubWebhook {
-	return &GitHubWebhook{
-		request: request,
+func NewGitHubWebhook(logger vcsclient.Log, endpoint string) *GitHubWebhook {
+	if endpoint == "" {
+		// Default to GitHub "Cloud"
+		endpoint = "https://github.com"
+	} else {
+		// For GitHub the API endpoint is https://api.github.com but the Web Interface URL is https://github.com
+		// So we remove the "api." prefix to the hostname
+		// Applied to Cloud and On-Prem versions of GitHub
+		endpoint = strings.Replace(endpoint, "://api.", "://", 1)
```

Comment:
Let's add a debug log entry after this parsing, to make it easier to troubleshoot issues